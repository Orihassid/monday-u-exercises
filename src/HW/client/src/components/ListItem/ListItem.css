

.new-item {
    display: flex;
    justify-content: space-between;
    list-style: none;
    width: 100%;
    margin-bottom: 15px;
    background-color: #f2f2f2;
    padding-left: 10px;
    height: 40px;
    align-items: center;
    animation: new-item-animation 0.3s linear forwards;
  }
  .new-item:hover{
    background-color: #4a8ee7;
  cursor: pointer;
  visibility: visible;
  }
  
  img{
  
    width: 40px;
    height: 40px;
    margin-bottom: 0%;
    margin-left: 30%;
    }
    .items{
      display: flex;
      align-items: center;
    }
    .inputText{
      padding: 0;
      outline: none;
      border: none;
    }
    .deleteButton:hover{
      cursor: pointer;
  
    }
    .editButton{
      margin-right: 10px;
     
    }
    .editButton:hover{
      cursor: pointer;
    }
    