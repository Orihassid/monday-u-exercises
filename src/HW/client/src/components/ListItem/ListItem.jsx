import { useState } from "react";
import "./ListItem.css";
import EditIcon from "@mui/icons-material/Edit";
import IconButton from "@mui/material/IconButton";
import DeleteIcon from "@mui/icons-material/Delete";
import SaveIcon from "@mui/icons-material/Save";
import { Checkbox } from "monday-ui-react-core";
import "monday-ui-react-core/dist/main.css";
import PropTypes from "prop-types";
export const ListItem = ({
  item,
  deleteItemAction,
  editItemNameAction,
  updateCheckBoxAction,
}) => {
  const [taskName, setTaskName] = useState(item.itemName);
  const [isEditClicked, setEditClicked] = useState(true);

  const handleCheckboxChange = async (e) => {
    try {
      await updateCheckBoxAction(item.itemId, e.target.checked);
    } catch (err) {
      throw new Error("failed to update status with checkbox");
    }
  };

  const handleEditButtonClick = () => {
    setEditClicked(false);
  };
  const handleSaveButtonClick = async () => {
    try {
      setEditClicked(true);
      const newTaskName = taskName.replace("catch", "");
      await editItemNameAction(item.itemId, newTaskName);
    } catch (err) {
      throw new Error("failed to edit task in db");
    }
  };
  const handleInputChange = (e) => {
    setTaskName(e.target.value);
  };

  const handleDeleteClick = async () => {
    await deleteItemAction(item.itemId);
  };

  return (
    <div>
      <li id={item.itemId} className="new-item">
        <div className="items">
          <Checkbox
            defaultChecked={item.status}
            onChange={handleCheckboxChange}
          />

          <input
            className="inputText"
            type="text"
            readOnly={isEditClicked}
            value={taskName}
            onChange={handleInputChange}
          />
          {item.isPokemon !== 0 && (
            <a>
              <img src={item.imageUrl} />
            </a>
          )}
        </div>
        <div>
          <IconButton
            aria-label="delete"
            size="large"
            color="error"
            onClick={handleDeleteClick}
          >
            <DeleteIcon className="deleteButton" fontSize="inherit" />
          </IconButton>
          {isEditClicked && (
            <IconButton
              onClick={handleEditButtonClick}
              aria-label="edit"
              size="large"
            >
              <EditIcon className="editIcon" fontSize="inherit" />
            </IconButton>
          )}
          {!isEditClicked && (
            <IconButton
              onClick={handleSaveButtonClick}
              aria-label="save"
              size="large"
            >
              <SaveIcon className="saveIcon" fontSize="inherit" />
            </IconButton>
          )}
        </div>
      </li>
    </div>
  );
};

ListItem.prototype = {
  item: PropTypes.object,
  deleteItemAction: PropTypes.func,
  editItemNameAction: PropTypes.func,
  updateCheckBoxAction: PropTypes.func,
};
export default ListItem;
