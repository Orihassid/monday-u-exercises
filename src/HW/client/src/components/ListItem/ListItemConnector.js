import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import ListItem from "./ListItem";
import{deleteItemAction,editItemNameAction,updateCheckBoxAction} from '../../redux/actions/itemsEntitiesActions'

const mapStateToProps = (state, ownProps) => {
     
    return {};
   
  };
  
  const mapDispatchToProps = (dispatch, ownProps) => {
    return bindActionCreators(
      {
        deleteItemAction,editItemNameAction,updateCheckBoxAction
       
      },
      dispatch
    );
    }  
    export default connect(mapStateToProps, mapDispatchToProps)(ListItem);