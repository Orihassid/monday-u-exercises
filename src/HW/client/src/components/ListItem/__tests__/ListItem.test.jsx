import renderer from "react-test-renderer";
import { ListItem } from "../ListItem";

const pendingTodo = {
  itemId: 1,
  itemName: "shopping",
  imageUrl: null,
  isPokemon: 0,
  pokemonId: null,
  status: 0,
};


test("renders correctly pending todo", () => {
  const tree = renderer
    .create(<ListItem item={pendingTodo}  key={pendingTodo.itemId} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly given wrong item props", () => {

    const item = {
        id: 3,
        text: "Pokemon",
     };
  const tree = renderer
    .create(<ListItem  item={item} key={item.id} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
