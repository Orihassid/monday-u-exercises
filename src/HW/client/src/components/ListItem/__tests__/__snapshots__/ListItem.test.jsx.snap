// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <li
    className="new-item"
  >
    <div
      className="items"
    >
      <label
        className="monday-style-checkbox"
        onClickCapture={[Function]}
        onMouseUp={[Function]}
      >
        <input
          className="monday-style-checkbox__input"
          defaultChecked={false}
          disabled={false}
          name=""
          onChange={[Function]}
          type="checkbox"
          value=""
        />
        <div
          className="monday-style-checkbox__checkbox monday-style-checkbox__prevent-animation"
        >
          <svg
            aria-hidden={true}
            className="icon_component monday-style-checkbox__icon icon_component--no-focus-style"
            fill="currentColor"
            height="16"
            onClick={[Function]}
            viewBox="0 0 20 20"
            width="16"
          >
            <path
              clipRule="evenodd"
              d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <span
          className="monday-style-checkbox__label"
        />
      </label>
      <input
        className="inputText"
        onChange={[Function]}
        readOnly={true}
        type="text"
      />
      <a>
        <img />
      </a>
    </div>
    <div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeLarge css-fz35w-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit deleteButton css-1vooibu-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="edit"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit editIcon css-1vooibu-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</div>
`;

exports[`renders correctly pending todo 1`] = `
<div>
  <li
    className="new-item"
    id={1}
  >
    <div
      className="items"
    >
      <label
        className="monday-style-checkbox"
        onClickCapture={[Function]}
        onMouseUp={[Function]}
      >
        <input
          className="monday-style-checkbox__input"
          defaultChecked={0}
          disabled={false}
          name=""
          onChange={[Function]}
          type="checkbox"
          value=""
        />
        <div
          className="monday-style-checkbox__checkbox monday-style-checkbox__prevent-animation"
        >
          <svg
            aria-hidden={true}
            className="icon_component monday-style-checkbox__icon icon_component--no-focus-style"
            fill="currentColor"
            height="16"
            onClick={[Function]}
            viewBox="0 0 20 20"
            width="16"
          >
            <path
              clipRule="evenodd"
              d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <span
          className="monday-style-checkbox__label"
        />
      </label>
      <input
        className="inputText"
        onChange={[Function]}
        readOnly={true}
        type="text"
        value="shopping"
      />
    </div>
    <div>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeLarge css-fz35w-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit deleteButton css-1vooibu-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </button>
      <button
        aria-label="edit"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit editIcon css-1vooibu-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </button>
    </div>
  </li>
</div>
`;
